### PR TITLE
Added multi-site parsing w2dynamics

### PR DESCRIPTION
**Pull request by JosePizarro3**
After discussing with the developers, I added the multi-site / multi-impurity case to the w2dynamics parser.

Now it is properly captured, but for that I had to tweek a bit the parsing of the `config_atoms_parameters`. Please @ladinesa can you check this? Thanks!

### CODE REVIEW

File: electronicparsers/w2dynamics/parser.py [hunk 1]
```diff
@@ -196,7 +202,7 @@ def parse_method(self, data):
         corr_orbs_per_atoms = []
         occ_per_atoms = []
         for i in range(sec_dmft.n_atoms_per_unit_cell):
-            corr_orbs_per_atoms.append(sec_method.x_w2dynamics_config.x_w2dynamics_config_atoms[i].x_w2dynamics_nd)
+            corr_orbs_per_atoms.append(sec_method.x_w2dynamics_config.x_w2dynamics_config_atoms[i].x_w2dynamics_nd + sec_method.x_w2dynamics_config.x_w2dynamics_config_atoms[i].x_w2dynamics_np)
```
Comment (ladinesa) [inline electronicparsers/w2dynamics/parser.py]: line break

File: electronicparsers/w2dynamics/parser.py [hunk 2]
```diff
@@ -172,18 +172,24 @@ def parse_method(self, data):
...
-                parameters = data.attrs.get(key)
-                keys_mod = (key.replace('-', '_')).split('.')
-                # TODO parse correctly sec_config_atoms as a repeat subsection
-                setattr(sec_config_subsection, f'x_w2dynamics_{keys_mod[-1]}', parameters)
+                if not key.startswith(f'atoms'):
+                    parameters = data.attrs.get(key)
+                    keys_mod = (key.replace('-', '_')).split('.')
+                    setattr(sec_config_subsection, f'x_w2dynamics_{keys_mod[-1]}', parameters)
+        # Parse Method.x_w2dynamics_config atoms quantities
+        for i in range(data.attrs.get(f'general.nat')):
```
Comment (ladinesa) [inline electronicparsers/w2dynamics/parser.py]: put a default 0 here

**Comment by ladinesa** [comment]
Only minor comments, please merge.

File: electronicparsers/w2dynamics/parser.py (hunk 2 above)
Comment (JosePizarro3) [reply electronicparsers/w2dynamics/parser.py]: you mean a 1?

Comment (ladinesa) [reply electronicparsers/w2dynamics/parser.py]: a zero inorder not to go through the loop.

File: electronicparsers/w2dynamics/parser.py (hunk 1 above)
Comment (JosePizarro3) [reply electronicparsers/w2dynamics/parser.py]: kind of messy the [W503](https://www.flake8rules.com/rules/W503.html) and the [W504](https://www.flake8rules.com/rules/W504.html) errors I am getting to this.

I think W503 should be ignored.

Comment (ladinesa) [reply electronicparsers/w2dynamics/parser.py]: understood. kinda hard to fix this.

File: electronicparsers/w2dynamics/parser.py (hunk 2 above)
Comment (JosePizarro3) [reply electronicparsers/w2dynamics/parser.py]: ok, I defined some `nd` and `np` in the lines before to make it more readable.

File: electronicparsers/w2dynamics/parser.py (hunk 1 above)
Comment (JosePizarro3) [reply electronicparsers/w2dynamics/parser.py]: Ok, in any case I added some `nd` and `np` in the lines before to make it more readable.

File: electronicparsers/w2dynamics/parser.py (hunk 2 above)
Comment (JosePizarro3) [reply electronicparsers/w2dynamics/parser.py]: done, I had to add also the default 0 in line 196

**Pull request merged**